Result/FormatResult.cs

```diff
@@ -14,74 +15,79 @@ public class FormatResult<T>
         private TreeNode<T> _tree;
         private ITracer _tracer;
 
-        public const string PathToJson = "\\JsonFormatter\\bin\\Debug\\JsonFormatter.dll";
+        public const string PathToJson = "\\JsonFormatter\\bin\\Debug\\JsonFormatter.dll"; 
         public const string PathToYaml = "\\YamlFormatter\\bin\\Debug\\YamlFormatter.dll";
         public const string PathToPlugins = "\\Plugins";
 
-
-
         public FormatResult(TreeNode<T> tree, ITracer tracer)
         {
             _tree = tree;
             _tracer = tracer;
         }
 
-        public void ToJson()
+        public void ToJson(string savePath)
         {
-            InvokePluginFormatter(PathToJson);
+            InvokePluginFormatter(PathToJson, savePath);
         }
 
-        public void ToConsole()
+        public void ToConsole(string output=null)
         {
-            _tree.Traverse(_tree, 0, true);
+            new FormatterConsole<T>().Format(_tree, _tracer, 0, true, output);
         }
 
-        public void ToXml()
+        public void ToXml(string savePath)
         {
-           new FormatterXml<T>().Format(_tree, _tracer, 0, true);
+           new FormatterXml<T>().Format(_tree, _tracer, 0, true, savePath);
         }
 
-        public void ToYaml()
+        public void ToYaml(string savePath)
         {
-            InvokePluginFormatter(PathToYaml);
+            InvokePluginFormatter(PathToYaml, savePath);
         }
-        //разобраться с xml
-        public string ToSpecialFormat(ICollection<IFormatter<T>> plugins, string formatName)
+       
+        public string ToSpecialFormat(ICollection<IFormatter<T>> plugins, string formatName, string output)
         {
+            string result = null;
             switch (formatName)
             {
                 case "console":
-                    ToConsole();
+                    ToConsole(output);
+                    result = formatName;
                     break;
                 case "xml":
-                    ToXml();
+                    ToXml(output);
+                    result = formatName;
                     break;
                 default:
                     string pathToPlugins = FormatResult<T>.GetAbsolutePath("Plugins");
+                    if (plugins.Count == 0)
```

Comment:
It's extra check. Please, remove.
If "plugins.Count == 0" it will be correctly processed by "foreach".